distribution/src/resources/config-tool/templates/conf/internal-apis.xml.j2

```diff
@@ -20,37 +20,86 @@
 -->
 <internalApis>
     <apis>
-        {% for api in internal_api %}
-        <api name="{{api.api_name}}" protocol="{{api.protocols}}" class="{{api.class}}"/>
-        {% endfor %}
-        <api name="{{management_api.name}}" protocol="{{management_api.protocols}}" class="{{management_api.class}}">
+        <api name="PrometheusApi" protocol="{{prometheus_api.protocols}}" class="org.wso2.micro.integrator.prometheus.publisher.service.MetricAPI">
+            {% if prometheus_api.basic_security_handler.enable == true
+                or prometheus_api.authorization_handler.enable == true or prometheus_api.custom_handlers is defined %}
+            <handlers>
+                {% if prometheus_api.basic_security_handler.enable == true %}
+                <handler class="org.wso2.micro.integrator.management.apis.security.handler.BasicSecurityHandler" name="BasicSecurityHandler"/>
+                {% endif %}
+                {% if prometheus_api.authorization_handler.enable == true %}
+                <handler class="org.wso2.micro.integrator.management.apis.security.handler.AuthorizationHandler" name="AuthorizationHandler">
+                    {% if prometheus_api.authorization_handler.resources is defined %}
+                    <resources>
+                        {% for resource in prometheus_api.authorization_handler.resources %}
+                        <resource>{{resource.path}}</resource>
+                        {% endfor %}
+                    </resources>
+                    {% endif %}
+                </handler>
+                {% endif %}
+                {% if prometheus_api.custom_handlers is defined %}
+                {% for handler in prometheus_api.custom_handlers %}
+                <handler name="{{handler.name}}" class="{{handler.class}}">
+                    {% if handler.resources is defined %}
+                    <resources>
+                        {% for resource in handler.resources %}
+                        <resource>{{resource.path}}</resource>
+                        {% endfor %}
+                    </resources>
+                    {% endif %}
+                </handler>
+                {% endfor %}
+                {% endif %}
+            </handlers>
+            {% endif %}
+        </api>
+        <api name="ReadinessProbe" protocol="http" class="org.wso2.micro.integrator.probes.ReadinessProbe"/>
+        <api class="org.wso2.micro.integrator.management.apis.ManagementInternalApi" name="ManagementApi" protocol="{{management_api.protocols}}">
             <handlers>
-                {% if management_api_token_handler.enable == true %}
-                <handler name="{{management_api.handler.name}}" class="{{management_api.handler.class}}">
+                {% if management_api.basic_security_handler.enable == true %}
+                <handler class="org.wso2.micro.integrator.management.apis.security.handler.BasicSecurityHandler" name="BasicSecurityHandler"/>
```

Comment:
The resources are supported for JWT and basic also by implementation isn't it . Are we just not exposing those to be configured .

Reply:
I thought of not adding it since it will allow unauthenticated access to the management api. Let's leave it as it is for now, and see if we get a requirement.